Common/src/geometry/CPhysicalGeometry.cpp

```diff
@@ -4346,431 +4346,305 @@ void CPhysicalGeometry::PrepareAdjacency(CConfig *config) {
 
 }
 
-void CPhysicalGeometry::Check_IntElem_Orientation(CConfig *config) {
-
-  unsigned long Point_1, Point_2, Point_3, Point_4, Point_5, Point_6,
-  iElem, triangle_flip = 0, quad_flip = 0, tet_flip = 0, prism_flip = 0,
-  hexa_flip = 0, pyram_flip = 0;
-  su2double test_1, test_2, test_3, test_4, *Coord_1, *Coord_2, *Coord_3, *Coord_4,
-  *Coord_5, *Coord_6, a[3] = {0.0,0.0,0.0}, b[3] = {0.0,0.0,0.0}, c[3] = {0.0,0.0,0.0}, n[3] = {0.0,0.0,0.0}, test;
-  unsigned short iDim;
-
-  /*--- Loop over all the elements ---*/
-
-  for (iElem = 0; iElem < nElem; iElem++) {
+void CPhysicalGeometry::Check_IntElem_Orientation(const CConfig *config) {
+
+  unsigned long tria_flip=0, quad_flip=0, tet_flip=0, prism_flip=0, hexa_flip=0, pyram_flip=0;
+  unsigned long quad_error=0, prism_error=0, hexa_error=0, pyram_error=0;
+
+  SU2_OMP_PARALLEL_(reduction(+:tria_flip,quad_flip,tet_flip,prism_flip,hexa_flip,pyram_flip)) {
+
+  /*--- Lambda to test triangles. Normal should be positive in the z direction (right hand rule). ---*/
+  auto checkTria = [&](unsigned long iElem, int Node_1, int Node_2, int Node_3) {
+    auto Coord_1 = nodes->GetCoord(elem[iElem]->GetNode(Node_1));
+    auto Coord_2 = nodes->GetCoord(elem[iElem]->GetNode(Node_2));
+    auto Coord_3 = nodes->GetCoord(elem[iElem]->GetNode(Node_3));
+    constexpr int nDim = 2;
+    su2double a[nDim]={0.0}, b[nDim]={0.0};
+    GeometryToolbox::Distance(nDim, Coord_2, Coord_1, a);
+    GeometryToolbox::Distance(nDim, Coord_3, Coord_1, b);
+    return a[0]*b[1]-a[1]*b[0] < 0.0;
+  };
+
+  /*--- Lambda to test tetrahedrons, volume must be positive,
+   * the normal of any face must point towards the other point. ---*/
+  auto checkTetra = [&](unsigned long iElem, int Node_1, int Node_2, int Node_3, int Node_4) {
+    auto Coord_1 = nodes->GetCoord(elem[iElem]->GetNode(Node_1));
+    auto Coord_2 = nodes->GetCoord(elem[iElem]->GetNode(Node_2));
+    auto Coord_3 = nodes->GetCoord(elem[iElem]->GetNode(Node_3));
+    auto Coord_4 = nodes->GetCoord(elem[iElem]->GetNode(Node_4));
+    constexpr int nDim = 3;
+    su2double a[nDim]={0.0}, b[nDim]={0.0}, c[nDim]={0.0}, n[nDim]={0.0};
+    GeometryToolbox::Distance(nDim, Coord_2, Coord_1, a);
+    GeometryToolbox::Distance(nDim, Coord_3, Coord_1, b);
+    GeometryToolbox::Distance(nDim, Coord_4, Coord_1, c);
+    GeometryToolbox::CrossProduct(a,b,n);
```

Comment:
If not too much trouble, I think it is more clear if all of the GeometryToolbox methods return outputs rather than using in/out params, e.g., something like
```
const std::array<su2double, nDim> a = GeometryToolbox::Distance(nDim, Coord_2, Coord_1);
```

Reply:
I encourage you to try that for yourself and see what the compiler tells you https://gcc.godbolt.org/